src/lib/replication/repl_dev/raft_repl_dev.cpp

```diff
@@ -166,176 +183,57 @@ void RaftReplDev::async_alloc_write(sisl::blob const& header, sisl::blob const&
             }
         });
     } else {
-        rreq->value_inlined = true;
-        RD_LOG(DEBUG, "Skipping data channel send since value size is 0");
+        RD_LOGD("Skipping data channel send since value size is 0");
+        rreq->add_state(repl_req_state_t::DATA_WRITTEN);
         auto raft_status = m_state_machine->propose_to_raft(rreq);
         if (raft_status != ReplServiceError::OK) { handle_error(rreq, raft_status); }
     }
 }
 
-void RaftReplDev::push_data_to_all_followers(repl_req_ptr_t rreq) {
-    auto& builder = rreq->fb_builder;
+void RaftReplDev::push_data_to_all_followers(repl_req_ptr_t rreq, sisl::sg_list const& data) {
```

Comment:
are there any logical change for all these functions we should pay attention in this file besides the rreq safety access?

Reply:
There is no logic change. Changes which is important is save_pushed_data and save_fetched_data, but there is no logic change there as well.